stream-rpc/src/subscriptions.rs

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    connection::Client,
+    counters,
+    errors::JsonRpcError,
+    json_rpc::{
+        Method, RequestIdentifier, SubscribeResult, SubscribeToEventsParams,
+        SubscribeToTransactionsParams,
+    },
+};
+use diem_json_rpc::data::{get_events, get_transactions};
+use diem_logger::{debug, warn};
+use rand::Rng;
+use std::{borrow::Borrow, sync::Arc};
+use storage_interface::DbReader;
+use tokio::task::JoinHandle;
+
+
+#[derive(Clone, Debug)]
+pub struct SubscriptionConfig {
+    pub fetch_size: u64,
+    pub poll_interval_ms: u64,
+    pub queue_size: u64,
+}
+
+/// Sleep for time_ms +/- rand(wiggle_ms or 10% of time_ms)
+pub async fn sleep_wiggled(time_ms: u64, wiggle_ms: Option<u64>) {
+    let time_to_sleep = time_ms
+        + rand::thread_rng().gen_range(0..wiggle_ms.unwrap_or((time_ms as f64 * 0.1) as u64));
+    tokio::time::sleep(tokio::time::Duration::from_millis(time_to_sleep)).await;
+}
+
+pub fn subscribe_to_transactions(
+    db: &Arc<dyn DbReader>,
+    client: Client,
+    jsonrpc_id: RequestIdentifier,
+    config: &Arc<SubscriptionConfig>,
+    params: &SubscribeToTransactionsParams,
+) -> Result<Option<JoinHandle<()>>, JsonRpcError> {
```

Comment:
It doesn't look like this function can return `Ok(None)`, maybe change the return type to `Result<JoinHandle<()>, JsonRpcError>`